polyfactory/utils/helpers.py

```diff
@@ -71,15 +65,14 @@ def unwrap_annotation(annotation: Any, random: Random) -> Any:
     :returns: The unwrapped annotation.
 
     """
-    while is_optional(annotation) or is_union(annotation) or is_new_type(annotation) or is_annotated(annotation):
+    while is_optional(annotation) or is_new_type(annotation) or is_annotated(annotation):
```

Comment:
Union handling is removed here as handled in `get_field_value` itself so redundant to handle twice. This removal may also fix issues seen in #366

Reply:
Once this is merged, I'll test it out to see if it resolves the issues in #366! If so, that'd be great :)